### PR TITLE
Add Canonical Service Dashboard

### PR DESCRIPTION
**Pull request by carolynhu**
https://github.com/istio/istio/issues/28288

This PR is doing two things:
(1) Add Canonical Service tables in Mesh Dashboard to replace existing K8s Service tables
(2) Add Canonical Service Dashboard

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by mandarjog** [comment]
Thanks Carolyn, can you add a rain tank link?

**Comment by carolynhu** [comment]
> Thanks Carolyn, can you add a rain tank link?

Yes, sure, I will add raintank link for both this and the "add canonical service dashboard" PR once the design doc review comments are resolved.

**Comment by carolynhu** [comment]
raintank link: https://snapshot.raintank.io/dashboard/snapshot/GxPLmBARJO5FaA2TXXGQlLJoYqlqCmyr

**Review by bianpengyuan — COMMENTED** [review]
lg, but shall we merge this after canonical service dashboard is in, or at least get them in together?

**Comment by carolynhu** [comment]
Canonical Service dashboard raintank: https://snapshot.raintank.io/dashboard/snapshot/auviY4jt4epXnlTmD9fhhju5AcQP6EKG

**Comment by carolynhu** [comment]
verified the link from "Canonical service tables" in Mesh dashboard to "Canonical service dashboard" works properly.

**Comment by bianpengyuan** [comment]
IMHO `canonical service` is more about an implementation detail and does not really make sense to most users. From user perspective, the dashboard is still just about `service`. Can we update the existing service dashboard query to something like `destination_canonical_service=svc OR destination_service=svc`.

**Comment by carolynhu** [comment]
> IMHO `canonical service` is more about an implementation detail and does not really make sense to most users. From user perspective, the dashboard is still just about `service`. Can we update the existing service dashboard query to something like `destination_canonical_service=svc OR destination_service=svc`.

As discussed in the design doc, @mandarjog wants use to deprecate the old service dashboard and add new CS dashboard. I think this is the formal way of doing it instead of modify things against the old service dashboard.

**Comment by bianpengyuan** [comment]
> I think this is the formal way of doing it instead of modify things against the old service dashboard.

`canonical service` is about service, and I think service dashboard is a good place to put it. I think we can either make the existing query flexible to accommodate both conditions, or add a completely new section or tab in the dashboard and mark the original ones deprecated.

**Comment by douglas-reid** [comment]
> > I think this is the formal way of doing it instead of modify things against the old service dashboard.
> 
> `canonical service` is about service, and I think service dashboard is a good place to put it. I think we can either make the existing query flexible to accommodate both conditions, or add a completely new section or tab in the dashboard and mark the original ones deprecated.

The two types of service dashboards will show different bits (workload traffic v. "service" traffic, etc). There doesn't seem to be a good way to display all the info on the same page without confusion and crowding.

In the future, I'd like to see this dashboard incorporate information about the workloads and k8s services that constitute the "canonical service".  If we do, we can preserve the old dashboard for a more k8s/infra level view. But that can follow on later, I imagine.

**Comment by carolynhu** [comment]
raintanks:
mesh dashboard: https://snapshot.raintank.io/dashboard/snapshot/BKyTlIqi3TNePbi9XHY3NLOkS1BKzYww
service dashboard: https://snapshot.raintank.io/dashboard/snapshot/QDMrRw5XdRpmbqDRTRnMRB3kPzsHDUrR

**Comment by carolynhu** [comment]
mesh dashboard: https://snapshot.raintank.io/dashboard/snapshot/UARC2Gmuj5dVxBUxlwWgsWJZ51glsIbP

service dashboard:
https://snapshot.raintank.io/dashboard/snapshot/b7SMPSPW5LihOYVsEpJyJ0SRBcOe8k8m
https://snapshot.raintank.io/dashboard/snapshot/44VBVKliKwjj0572E3RNGnoZQ3JVFktf

**Comment by douglas-reid** [comment]
Looking good. Thanks for continuing to work on this.

A few questions:
- curious: is it possible (and/or easy) to combine the `Service` and `Service Namespace` columns in the mesh dashboard (to match the workload column)?
- have we tried the service dashboard with external services yet ?

File: manifests/addons/dashboards/istio-mesh-dashboard.json [hunk 1]
```diff
@@ -19,14 +19,15 @@
...
+      "collapsed": false,
+      "content": "<div>\n  <div style=\"position: absolute; bottom: 0\">\n    <a href=\"https://istio.io\" target=\"_blank\" style=\"font-size: 30px; text-decoration: none; color: inherit\"><img src=\"https://istio.io/latest/img/istio-bluelogo-nobackground-unframed.svg\" style=\"height: 50px\"> Istio</a>\n  </div>\n  <div style=\"position: absolute; bottom: 0; right: 0; font-size: 15px\">\n    Istio is an <a href=\"https://github.com/istio/istio\" target=\"_blank\">open platform</a> that provides a uniform way to connect,\n    <a href=\"https://istio.io/latest/docs/concepts/traffic-management/\" target=\"_blank\">manage</a>, and \n    <a href=\"https://istio.io/latest/docs/concepts/security/\" target=\"_blank\">secure</a> microservices.  Need help? Join the <a href=\"https://istio.io/community/\" target=\"_blank\">Istio community</a>.\n  <br> See changes of Istio Grafana Dashboards, please check out our <a href=\"https://github.com/istio/istio/releasenotes/notes/canonical-service-dashboard.yaml\" target=\"_blank\">Release Notes</a>",
```
Comment (douglas-reid) [inline manifests/addons/dashboards/istio-mesh-dashboard.json:23]:
```suggestion
      "content": "<div>\n  <div style=\"position: absolute; bottom: 0\">\n    <a href=\"https://istio.io\" target=\"_blank\" style=\"font-size: 30px; text-decoration: none; color: inherit\"><img src=\"https://istio.io/latest/img/istio-bluelogo-nobackground-unframed.svg\" style=\"height: 50px\"> Istio</a>\n  </div>\n  <div style=\"position: absolute; bottom: 0; right: 0; font-size: 15px\">\n    Istio is an <a href=\"https://github.com/istio/istio\" target=\"_blank\">open platform</a> that provides a uniform way to connect,\n    <a href=\"https://istio.io/latest/docs/concepts/traffic-management/\" target=\"_blank\">manage</a>, and \n    <a href=\"https://istio.io/latest/docs/concepts/security/\" target=\"_blank\">secure</a> microservices.  Need help? Join the <a href=\"https://istio.io/community/\" target=\"_blank\">Istio community</a> or browse our <a href=\"https://github.com/istio/istio/releasenotes/notes/canonical-service-dashboard.yaml\" target=\"_blank\">release notes</a>",
```

File: releasenotes/notes/canonical-service-dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
+issue:
+  - 28288
+releaseNotes:
+  - |
+    **Added** In Istio release 1.9, we replace the Istio Service Dashboard from all the queries are based on the K8s service to Istio Canonical Service. The benefits of this change is that user now can gain service-to-service traffic/metrics knowledge in a much more Istio friendly way.
```
Comment (douglas-reid) [inline releasenotes/notes/canonical-service-dashboard.yaml:8]:
```suggestion
    **Updated** the Istio Service Dashboard to use canonical services. The updated dashboard will now show service-to-service traffic for both incoming and outgoing links. More information on canonical services and their role in Istio is provided <link>.
```

File: samples/addons/README.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ You can use Grafana to monitor the health of Istio and of applications within th
 This sample provides the following dashboards:
 
 * [Mesh Dashboard](https://grafana.com/grafana/dashboards/7639) provides an overview of all services in the mesh.
-* [Service Dashboard](https://grafana.com/grafana/dashboards/7636) provides a detailed breakdown of metrics for a service.
+* [Service Dashboard](https://grafana.com/grafana/dashboards/7636) provides a detailed breakdown of metrics for an Istio Canonical service.
```
Comment (douglas-reid) [inline samples/addons/README.md:42]: I think we can just leave this as it was

**Comment by carolynhu** [comment]
> * curious: is it possible (and/or easy) to combine the `Service` and `Service Namespace` columns in the mesh dashboard (to match the workload column)?

i tried it, it's not easy to combine them, since we have workload involved, which Workload column name is generated from {{ destination_workload}}.{{ destination_workload_namespace }}

we do not have a good place to put for {{ destination_canonical_service }}.{{ destination_workload_namespace }}

**Comment by carolynhu** [comment]
> * have we tried the service dashboard with external services yet ?

I'm following this https://istio.io/latest/docs/tasks/traffic-management/egress/egress-control/ and tried the sleep.yaml app, since we are using the destination_canonical_service label for now, those are `unknown` in this case. Which means we could not gather any data related to that yet.

```
istio_requests_total{app="sleep",connection_security_policy="unknown",destination_app="unknown",destination_canonical_revision="latest",destination_canonical_service="unknown",destination_principal="unknown",destination_service="httpbin.org",destination_service_name="httpbin.org",destination_service_namespace="unknown",destination_version="unknown",destination_workload="unknown",destination_workload_namespace="unknown",instance="10.56.1.9:15020",istio_io_rev="default",job="kubernetes-pods",kubernetes_namespace="default",kubernetes_pod_name="sleep-dd98b5f48-j8r9b",pod_template_hash="dd98b5f48",reporter="source",request_protocol="http",response_code="200",response_flags="-",security_istio_io_tlsMode="istio",service_istio_io_canonical_name="sleep",service_istio_io_canonical_revision="latest",source_app="sleep",source_canonical_revision="latest",source_canonical_service="sleep",source_principal="unknown",source_version="unknown",source_workload="sleep",source_workload_namespace="default"}
```
as you can see from: 
https://snapshot.raintank.io/dashboard/snapshot/3fTndwb6PCZlTfFSVMjawddEcMO34pqI
https://snapshot.raintank.io/dashboard/snapshot/rMwsq6nAZY6RGqfFT1Rdo5rZULqglXij

**Pull request closed without merging**